azure-devops/run-build.yml

```diff
@@ -19,12 +19,52 @@ jobs:
       displayName: 'Setup TMP Directory'
     - checkout: self
       clean: true
-      submodules: true
+      submodules: false
+    - task: PowerShell@2
+      displayName: 'Get submodule SHAs'
+      timeoutInMinutes: 1
+      inputs:
+        targetType: inline
+        script: |
+          cd $(Build.SourcesDirectory)
+          $pattern = "[^a-zA-Z0-9]"
+          $llvmSHA = git submodule status llvm-project | %{($_ -split "\s+")[0]}
+          $llvmSHA = $llvmSHA -replace $pattern,""
+          Write-Host "##vso[task.setvariable variable=llvmSHA;]$llvmSHA"
+          $vcpkgSHA = git submodule status vcpkg | %{($_ -split "\s+")[0]}
+          $vcpkgSHA = $vcpkgSHA -replace $pattern,""
+          Write-Host "##vso[task.setvariable variable=vcpkgSHA;]$vcpkgSHA"
+    - script: |
+        cd $(Build.SourcesDirectory)
+        if not exist "llvm-project" (
+          mkdir llvm-project
+        )
+        cd llvm-project
+        git init
+        git remote add llvm https://github.com/llvm/llvm-project

```

Comment:
Won't this fail the second time a job runs on this server? I get `fatal: remote llvm already exists.` with exit code 128 when I recreate this scenario locally. Do script steps not fail when one of their command lines returns a non-zero exit status?

Reply:
Yup scripts only fail if the last command is non zero

Reply:
That seems confusing for logging purposes, although I don't exactly object.